File: Data Structures & Algorithms/remove-element/submission-1.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} val
     * @return {number}
     */
    removeElement(nums, val) {
        let right = nums.length - 1;
        let left = 0;
        let k = 0;
        while (right > left) {
            if (nums[left] === val) {
                if (nums[right] !== val) {
                    nums[left++] = nums[right]
                }
                right--;
            } else {
                left++;
            }
            k++;
        }
            console.log(k);
        if (nums.length % 2 !== 0) {
            k++;
        }
        return k;
    }
}

File: Data Structures & Algorithms/special-array-i/submission-1.cs
public class Solution {
    public bool IsArraySpecial(int[] nums) {
        if (nums.Length == 1) {
            return true;
        }

        if (nums.Length % 2 == 0) {
            if ((nums[0] % 2 == 0) && !(nums[nums.Length - 1] % 2 == 1)) {
                return false;
            }
            if ((nums[0] % 2 == 1) && !(nums[nums.Length - 1] % 2 == 0)) {
                return false;
            }
        }

        if (nums.Length % 2 == 1) {
            if ((nums[0] % 2 == 0) && (!(nums[nums.Length - 1] % 2 == 0))) {
                return false;
            }
            if ((nums[0] % 2 == 1) && !(nums[nums.Length - 1] % 2 == 1)) {
                return false;
            }
        }

        int left = 1;
        int right = nums.Length - 2;
        bool isEven = nums.Length % 2 == 0;

        while(left < right) {
            int[] arr = {nums[left], nums[right]};
            if (isEven) {
                if ((arr[0] % 2 == 0) && !(arr[arr.Length - 1] % 2 == 1)) {
                    return false;
                }
                if ((arr[0] % 2 == 1) && !(arr[arr.Length - 1] % 2 == 0)) {
                    return false;
                }
            }

            if (!isEven) {
                if (arr[0] % 2 == 0 && arr[arr.Length - 1] % 2 == 0) {
                    return false;
                }
                if (arr[0] % 2 == 1 && arr[arr.Length - 1] % 2 == 1) {
                    return false;
                }
            }
            left++;
            right--;
        }
        return true;
    }
}